src/test/mod.rs

```diff
@@ -133,3 +133,48 @@ fn basic_where_clauses_fail() {
                                    expression evaluated to an empty collection: `decls.trait_invariants()`"#]]
     )
 }
+
+#[test]
+fn trait_items_with_duplicate_fn_names() {
+    crate::assert_err!(
+        [
+            crate core {
+                trait A {
+                    fn a() -> ();
+                    fn a() -> ();
+                }
+            }
+        ]
+
+        [ /* TODO */ ]
```

Comment:
```suggestion
        [ "function name `a` is defined multiple times" ]
```

try that, I think

Reply:
Have updated accordingly on this commit:  1cff4c250d9d7acfe95102a96070724d01b88757.